meteor-app/imports/ui/assessment/assessment-add.js

```diff
@@ -55,17 +57,97 @@ class AssessmentAdd extends Component {
       this.props.resetId()
   }
 
-  onSubmit = ({ formData }) => {
-    const lastStep = schemas.length == this.state.step
+  onSubmit = async ({ formData }) => {
+    const lastStep = this.state.step == 4
+
+    const totalServiceCost = this.props.services
+      .map(key => {
+        if (!formData) { return 0 }
+        return formData.services.includes(key.name) ? key.price : 0
+      })
+      .reduce((a, b) => a + b) || 0
+    const totalPartsCost = this.props.serviceItems
+      .map(key => {
+        if (!formData) { return 0 }
+        return formData.parts.includes(key.name) ? key.price : 0
+      })
+      .reduce((a, b) => a + b) || 0
+    const totalCost = totalServiceCost + totalPartsCost + (formData ? formData.additionalFee * 100 || 0 : 0)
+        
     if (lastStep) {
-      this.props.setAssessment(this.state.formData)
+      const serviceItem = this.props.services
+        .filter(key => {
+          return formData.services.includes(key.name)
+        })
+        .map(key => {
+          return {
+            name: key.name, 
+            price: key.price, 
+            package: key.package
+          }
+        })
+      const partsItem = this.props.serviceItems
+        .filter(key => {
+          return formData.parts.includes(key.name)
+        })
+        .map(key => {
+          return {
+            name: key.name, 
+            price: key.price
+          }
+        })
+      const search = formData.b2bRefurbish ? 
+        'Refurbished Bike' : 
+        (formData.name + formData.email + formData.bikeMake + formData.bikeColor)
+      // Structuring form submission to match collection schema
+      const formResult = {
+        customerDetails: {
+          name: formData.name || '',
```

Comment:
I don't see the need to supply default values on the last step - surely these came from the form as you stepped through the wizard? To me this looks like a whole bunch of unnecessary code?